dnf5daemon-server/services/goal/goal.cpp

```diff
@@ -342,10 +342,32 @@ sdbus::MethodReply Goal::get_transaction_problems(sdbus::MethodCall & call) {
     return reply;
 }
 
+/// Returns true in case all inbound packages in the transaction comes from
+/// repository with gpg checks enforced.
+bool is_transaction_trusted(libdnf5::base::Transaction * transaction) {
+    for (const auto & tspkg : transaction->get_transaction_packages()) {
+        if (transaction_item_action_is_inbound(tspkg.get_action())) {
+            auto repo = tspkg.get_package().get_repo();
+            if (repo->get_type() == libdnf5::repo::Repo::Type::COMMANDLINE) {
+                // local rpm file installation is untrusted
```

Comment:
I heard rumors that RPM version 6 is going to enforce signatures <https://lists.fedoraproject.org/archives/list/devel@lists.fedoraproject.org/message/K4SO2G63Z7EGLGTXKFQX52SNZ45372V7/>. That's maybe why Panu sent a patch #2232 for DNF5.

I don't mind this conservative approach. (It's maybe good as an proactive measure against tricking a user into downloading a package with a web browser and then pointing the D-Bus client onto the local RPM file.) I only say that your pul request could be unnecessary strict.

Reply:
Thanks for pointing that out. For the time being, I'd rather stay on the safe side, but I'll definitely be monitoring the development in RPM.